tests/ops/test_flash_attn_varlen.py

```diff
@@ -0,0 +1,279 @@
+import math
+import os
+from typing import Optional, Tuple
+import torch.nn.functional as F
+import einops
+import pytest
+import torch
+import torchacc as ta
+from torch import nn
+
+from flash_attn import flash_attn_func, flash_attn_varlen_func
+from flash_attn.bert_padding import index_first_axis, pad_input, unpad_input  # noqa
+
+
+def _get_unpad_data(attention_mask):
+    seqlens_in_batch = attention_mask.sum(dim=-1, dtype=torch.int32)
+    indices = torch.nonzero(attention_mask.flatten(), as_tuple=False).flatten()
+    max_seqlen_in_batch = seqlens_in_batch.max().item()
+    cu_seqlens = F.pad(
+        torch.cumsum(seqlens_in_batch, dim=0, dtype=torch.int32), (1, 0))
+    return (
+        indices,
+        cu_seqlens,
+        max_seqlen_in_batch,
+    )
+
+
+class FlashAttention2(nn.Module):
```

Comment:
What does the FlashAttention2 mean? Does FA2 stand for it, or is the 2 just a magic number?

Reply:
Just a magic number, removed